modules/create_timepoint/php/ajaxtimepoint.class.inc

```diff
@@ -0,0 +1,280 @@
+<?php declare(strict_types=1);
+/**
+ * POST request methods.
+ *
+ * Used to retrieve info & create timepoint by the create_timepoint form.
+ * Ensures the user can create timepoint before processing
+ * the POST body (array) data.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Create_Timepoint
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\create_timepoint;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Create_Timepoint
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class AjaxTimepoint extends \NDB_Page
+{
+    public $skipTemplate = true;
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+            case 'GET':
+                return $this->_handleGET($request);
+            case 'POST':
+                return $this->_handlePOST($request);
+            default:
+                return new \LORIS\Http\Response\MethodNotAllowed(
+                    $this->allowedMethods()
+                );
+        }
+    }
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+        // Verify identifier from Post.
+        $values['identifier'] = isset($values['identifier']) ?
+            intval($values['identifier']) : null;
+        // User permission verification.
+        $values = $this->_verifyPermissions($request, $values);
```

Comment:
I think it makes more sense to create private class variables rather than pass `$values` all over the place in this class.

Reply:
@johnsaigle I kind of prefer passing the $values here. I'm sort of against class variables here because I find them less intuitive to keep track of the process and when the function returns the new set of values.